### PR TITLE
fix(pattern): validate and do not pass cta into cardsectionimages

### PR DESCRIPTION
**Pull request by mkothur**
### Related Ticket(s)

Related issue: #1196 

### Description

validating the otherProps and making sure don’t pass it into the CardSectionImages component. 

### Changelog

 - remove specific property from object of spread syntax.

### CODE REVIEW

File: packages/react/src/patterns/sections/CardSectionImages/__stories__/CardSectionImages.stories.js [hunk 1]
```diff
@@ -20,11 +20,18 @@ storiesOf('Patterns (Sections)|CardSectionImages', module)
...
+      cta: {
+        href: 'https://www.example.com',
+      },
+    };
     return (
       <CardSectionImages
         heading={text('Heading (required):', 'Read more about it')}
         theme={select('theme', themes, themes.white)}
         cards={object('Data', cards.CardSectionImages)}
+        cta={cta}
```
Comment (photodow) [inline packages/react/src/patterns/sections/CardSectionImages/__stories__/CardSectionImages.stories.js]: Shouldn't we not be passing this in here?

File: packages/react/src/patterns/sections/CardSectionImages/CardSectionImages.js [hunk 2]
```diff
@@ -21,7 +21,8 @@ const CardSectionImages = ({ cards, ...otherProps }) => {
     ({ image, eyebrow, heading, copy, cta: { href } }) =>
       image && eyebrow && heading && !copy && href
   );
-  return <CardSection {...otherProps} cards={cardsWithImages} />;
+  const { ['cta']: _, ...newObj } = otherProps;
```
Comment (photodow) [inline packages/react/src/patterns/sections/CardSectionImages/CardSectionImages.js]: This is written out in a confusing way. Is there anyreason why we couldn't just use `delete`?

```
delete otherPropse.cta;
return <CardSection {...otherProps} cards={cardsWithImages} />;
```

File: packages/react/src/patterns/sections/CardSectionImages/__stories__/CardSectionImages.stories.js (hunk 1 above)
Comment (mkothur) [reply packages/react/src/patterns/sections/CardSectionImages/__stories__/CardSectionImages.stories.js]: Ok will remove.

File: packages/react/src/patterns/sections/CardSectionImages/CardSectionImages.js (hunk 2 above)
Comment (mkothur) [reply packages/react/src/patterns/sections/CardSectionImages/CardSectionImages.js]: I also thought the delete operator, but I may get any comments from you on that delete operator , so tried another syntax to remove specific prop from spread operator.

**Comment by mkothur** [comment]
@photodow updated the PR.

**Comment by jeffchew** [comment]
> This works for me. Not sure if @jeffchew or anyone else has concerns about using `delete` here though. I'd love to get their thoughts on it.

I think should be fine.

**Pull request merged**